plugins/module_utils/foreman_helper.py

```diff
@@ -453,8 +454,13 @@ def _update_entity(self, resource, desired_entity, current_entity, params, entit
         desired_entity = _flatten_entity(desired_entity, entity_spec)
         current_entity = _flatten_entity(current_entity, entity_spec)
         for key, value in desired_entity.items():
-            if current_entity.get(key) != value:
-                payload[key] = value
+            # String comparison needs extra care in face of unicode
```

Comment:
can we have a test for this? or is the `architecture_name` below testing this?

Reply:
Well, we can ask for more, but the strangely appearing new processor architecture i invented there tests it.
To be more specific: On python3 it worked without this, on python2 it failed idempotence, because it compared bytes to unicode.

Reply:
that works for me.